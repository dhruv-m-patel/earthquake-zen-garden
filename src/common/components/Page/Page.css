@import '../../styles/variables.css';

html {
  font-size: 100%;
  height: 100vh;
}

body {
  margin: 0;
  padding: 0;
  font-size: var(--baseFontSize);
  font-family: var(--baseFontFamily);
  min-height: 100vh;
  color: var(--black);
  background-color: var(--white);

  & a {
    color: var(--blue);
  }

  & a:hover {
    text-decoration: underline;
    cursor: pointer;
  }
}

.appHeader {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
}
