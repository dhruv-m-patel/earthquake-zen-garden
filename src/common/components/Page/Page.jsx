import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames/bind';
import DefaultHelmet from '../DefaultHelmet';
import styles from './Page.css';

const cx = classnames.bind(styles);

export default function Page({ title, children }) {
  return (
    <div className={cx('app')}>
      <DefaultHelmet title={title} />
      <header className={cx('appHeader')}>
        <h2>Earthquake Zen Garden</h2>
      </header>
      <br />
      <br />
      <div className={cx('content')}>{children}</div>
    </div>
  );
}

Page.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string,
};

Page.defaultProps = {
  title: undefined,
};
