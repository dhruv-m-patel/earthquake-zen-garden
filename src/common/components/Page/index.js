import { connect } from 'react-redux';
import { fetchConfiguration } from '../../../client/redux/actions';
import Page from './Page';

function mapStateToProps({ config }) {
  return {
    isFetchingConfiguration: config.isFetching,
    config: config.data,
    error: config.error,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    fetchSiteConfiguration: () => dispatch(fetchConfiguration()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Page);
