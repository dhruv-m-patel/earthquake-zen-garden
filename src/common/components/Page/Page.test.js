import React from 'react';
import Tester from '../../../../tests/Tester';
import Page from './Page';

const tester = new Tester();

describe('Page', () => {
  test('it should render', () => {
    const snapshot = tester.getSnapshot(Page, {
      title: 'Test App',
      children: <div />,
    });
    expect(snapshot).toMatchSnapshot();
  });
});
