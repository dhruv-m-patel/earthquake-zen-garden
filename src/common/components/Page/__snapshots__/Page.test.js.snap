// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page it should render 1`] = `
<div
  className="Page__app___1b3SY"
>
  <header
    className="Page__appHeader___zKEHC"
  >
    <h2>
      Earthquake Zen Garden
    </h2>
  </header>
  <br />
  <br />
  <div
    className="Page__content___3sLpF"
  >
    <div />
  </div>
</div>
`;
